sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -1450,7 +1450,7 @@ ProgramManager::ProgramManager()
 
 const char *getArchName(const device &Device) {
   namespace syclex = sycl::ext::oneapi::experimental;
-  auto Arch = getSyclObjImpl(Device)->getDeviceArch();
+  const auto &Arch = getSyclObjImpl(Device)->getDeviceArch();
```

Comment:
This is not related to `getSyclObjImpl` and also it is not a functional change since `getDeviceArch` returns by value.